### PR TITLE
Rubocop update

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: ytools/y2tool/rubocop_yast_style.yml [hunk 1]
```diff
@@ -88,9 +113,63 @@ Style/CaseIndentation:
...
-# allow more than 10 lines for methods
+# allow more than 10 lines for methods as some team member feel unconfortable with it
 Metrics/MethodLength:
   Max: 30
 
+# allow more than 100 lines for class as some team member feel unconfortable with it
 Metrics/ClassLength:
   Max: 250
+
+# YaST use older rubocop that do not have allow predicates as default to true,
```
Comment (imobach) [inline ytools/y2tool/rubocop_yast_style.yml]: `does not allow` ?

**Comment by kobliha** [comment]
Nice :)

File: ytools/y2tool/rubocop_yast_style.yml [hunk 2]
```diff
@@ -88,9 +113,63 @@ Style/CaseIndentation:
...
+# so explicitelly allow it for older ones
```
Comment (imobach) [inline ytools/y2tool/rubocop_yast_style.yml]: `explicitly`

File: ytools/y2tool/rubocop_yast_style.yml [hunk 3]
```diff
@@ -88,9 +113,63 @@ Style/CaseIndentation:
...
+ AllowPredicates: true
+
+# Allow using and/or for driving code flow as its original intention.
+# Forbid it only in conditionals
+# So this rule allow something like
+# a = action_a or raise "Cannot do a"
+Style/AndOr:
+  EnforcedStyle: conditionals
+
+# Access modified afect globally all following method definition, so
```
Comment (imobach) [inline ytools/y2tool/rubocop_yast_style.yml]: `affect`

**Comment by imobach** [comment]
Just minor NP. Otherwise, LGTM.

**Comment by ancorgs** [comment]
LGTM, I agree with all the changes.

File: ytools/y2tool/rubocop_yast_style.yml [hunk 4]
```diff
@@ -88,9 +113,63 @@ Style/CaseIndentation:
...
+# it deserve bigger visibility then hiddin in method definition block.
+# for that reason we found better style:
+# class C
+#   trash
```
Comment (lslezak) [inline ytools/y2tool/rubocop_yast_style.yml]: NP: Use just one line with `...`, this comment is too long IMO.

**Comment by lslezak** [comment]
LGTM, I agree with the changes, thanks for updating the config! :+1:

File: ytools/y2tool/rubocop_yast_style.yml [hunk 5]
```diff
@@ -88,9 +113,63 @@ Style/CaseIndentation:
...
+#   ...
+#
+#   private
+#
+#   ....
+#   trash
+#   ...
+# end
+Style/AccessModifierIndentation:
+  EnforcedStyle: outdent
```
Comment (mvidner) [inline ytools/y2tool/rubocop_yast_style.yml]: So this is the one change that will make other packages fail, right? Will you fix them all?

Comment (jreidinger) [inline ytools/y2tool/rubocop_yast_style.yml]: yes, but it is not all packages, as two have exception, few do not use access modificators at all and few will fail. I will do it together with removing integrated exceptions.

File: ytools/y2tool/rubocop_yast_style.yml [hunk 6]
```diff
@@ -69,6 +69,9 @@ Style/SignalException:
   StyleGuide: https://github.com/bbatsov/ruby-style-guide#fail-method
   EnforcedStyle: only_raise
 
+Style/NumericLiterals:
+  StyleGuide: https://github.com/SUSE/style-guides/blob/master/Ruby.md#stylenumericliterals
+  Enabled: false
```
Comment (mvidner) [inline ytools/y2tool/rubocop_yast_style.yml]: > \# Do not require an underscore each three digits. Both 65536 and 65_536 are allowed.

File: ytools/y2tool/rubocop_yast_style.yml [hunk 7]
```diff
@@ -88,9 +113,63 @@ Style/CaseIndentation:
...
   Max: 30
 
+# allow more than 100 lines for class as some team member feel unconfortable with it
 Metrics/ClassLength:
   Max: 250
+
+# YaST use older rubocop that do not have allow predicates as default to true,
+# so explicitelly allow it for older ones
+Style/TrivialAccessors:
+ # predicates cannot be done with accessors
```
Comment (mvidner) [inline ytools/y2tool/rubocop_yast_style.yml]:
```rb
# Offense, a trivial accessor:
#    def favorite_color; @favorite_color; end
# No offense, a predicate:
#    def unladen_swallow?; @unladen_swallow; end
```

**Comment by jreidinger** [comment]
no opposition so lets merge it.

**Pull request merged**